File: aon-occam/src/main/java/com/esferalia/aon/occam/api/model/task/DailyTracking.java
package com.esferalia.aon.occam.api.model.task;

import java.io.Serializable;
import java.util.Date;
import java.util.Objects;

import com.esferalia.aon.occam.api.model.Domain;
import com.esferalia.aon.occam.api.model.registry.Project;
import com.esferalia.aon.occam.api.model.registry.Registry;

public class DailyTracking  implements Serializable{
	
	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;
	
	private Integer id;
	private Domain domain; 
	private TaskHolder taskHolder; 
	private Date trackingDate; 
	private Double trackingDuration; 
	private Integer jobType;
	private Registry registry; 
	private Project project; 
	private Integer activityType;
	private String comments; 
	private Integer task;
	private Double cost; 
	
	public DailyTracking() { 
	}

	public Integer getId() {
		return id;
	}

	public DailyTracking setId(Integer id) {
		this.id = id;
		return this;
	}

	public Domain getDomain() {
		if(domain == null) {
			domain = new Domain();
		}
		return domain;
	}

	public DailyTracking setDomain(Domain domain) {
		this.domain = domain;
		return this;
	}
	
	public TaskHolder getTaskHolder() {
		if(taskHolder == null) {
			taskHolder = new TaskHolder();
		}
		return taskHolder;
	}

	public DailyTracking setTaskHolder(TaskHolder taskHolder) {
		this.taskHolder = taskHolder;
		return this;
	}
	
	public Date getTrackingDate() {
		return trackingDate;
	}

	public DailyTracking setTrackingDate(Date trackingDate) {
		this.trackingDate = trackingDate;
		return this;
	}
	
	public Double getTrackingDuration() {
		return trackingDuration;
	}

	public DailyTracking setTrackingDuration(Double trackingDuration) {
		this.trackingDuration = trackingDuration;
		return this;
	}

	public Integer getJobType() {
		return jobType;
	}

	public DailyTracking setJobType(Integer jobType) {
		this.jobType = jobType;
		return this;
	}
	
	public Registry getRegistry() {
		return registry;
	}

	public DailyTracking setRegistry(Registry registry) {
		this.registry = registry;
		return this;
	}
	
	public Project getProject() {
		return project;
	}

	public DailyTracking setProject(Project project) {
		this.project = project;
		return this;
	}

	public Integer getActivityType() {
		return activityType;
	}

	public DailyTracking setActivityType(Integer activityType) {
		this.activityType = activityType;
		return this;
	}

	public String getComments() {
		return comments;
	}

	public DailyTracking setComments(String comments) {
		this.comments = comments;
		return this;
	}
	
	public Integer getTask() {
		return task;
	}

	public DailyTracking setTask(Integer task) {
		this.task = task;
		return this;
	}
	
	public Double getCost() {
		return cost;
	}

	public DailyTracking setCost(Double cost) {
		this.cost = cost;
		return this;
	}
	

	@Override
	public int hashCode() {
		return Objects.hashCode(id);
	}
	
	@Override
	public boolean equals(Object obj) {
		if (!(obj instanceof DailyTracking ) )
			return false;
		
		DailyTracking dailyTracking = (DailyTracking) obj;
		
		return Objects.equals(id, dailyTracking.id);
	}
}

File: aon-solutions/src/main/webapp/models/task/DailyTracking.js
import { Domain } from "../Domain.js";
import { Project } from "../project/Project.js";
import { TaskHolder } from "../project/TaskHolder.js";
import { Registry } from "../registry/Registry.js";

export class DailyTracking {
  id;
  domain;
  task_holder;
  tracking_date;
  tracking_duration;
  job_type;
  registry;
  project;
  activity_type;
  comments;
  task;
  cost;

  constructor(dailyTracking) {
    if (dailyTracking) {
      this.setDailyTracking(dailyTracking);
    } else {
      this.id = undefined;
      this.domain = new Domain();
      this.task_holder = new TaskHolder();
      this.tracking_date = undefined;
      this.tracking_duration = undefined;
      this.job_type = undefined;
      this.registry = new Registry();
      this.project = new Project();
      this.activity_type = undefined;
      this.comments = undefined;
      this.task = undefined;
      this.cost = undefined;
    }
  }

  setDailyTracking(dailyTracking) {
    if (dailyTracking) {
      this.setId(dailyTracking.id || undefined);
      this.setDomain(new Domain(dailyTracking.domain));
      this.setTaskHolder(new TaskHolder(dailyTracking.task_holder));
      this.setTrackingDate(dailyTracking.tracking_date || undefined);
      this.setTrackingDuration(dailyTracking.tracking_duration || undefined);
      this.setJobType(dailyTracking.job_type || undefined);
      this.setRegistry(new Registry(dailyTracking.registry));
      this.setProject(new Project(dailyTracking.project));
      this.setActivityType(dailyTracking.activity_type || undefined);
      this.setComments(dailyTracking.comments || undefined);
      this.setTask(dailyTracking.task || undefined);
      this.setCost(dailyTracking.cost || undefined);
    }
  }

  setId(id) {
    this.id = id;
    return this;
  }
  
  setDomain(domain) {
    this.domain = domain;
    return this;
  }

  setTaskHolder(taskHolder) {
    this.task_holder = taskHolder;
    return this;
  }

  setTrackingDate(trackingDate) {
    this.tracking_date = trackingDate;
    return this;
  }

  setTrackingDuration(trackingDuration) {
    this.tracking_duration = trackingDuration;
    return this;
  }

  setJobType(jobType) {
    this.job_type = jobType;
    return this;
  }

  setRegistry(registry) {
    this.registry = registry;
    return this;
  }

  setProject(project) {
    this.project = project;
    return this;
  }

  setActivityType(activityType) {
    this.activity_type = activityType;
    return this;
  }

  setComments(comments) {
    this.comments = comments;
    return this;
  }

  setTask(task) {
    this.task = task;
    return this;
  }

  setCost(cost) {
    this.cost = cost;
    return this;
  }

  getId() {
    return this.id;
  }

  getDomain() {
    return this.domain;
  }

  getTaskHolder() {
    return this.task_holder;
  }

  getTrackingDate() {
    return this.tracking_date;
  }

  getTrackingDuration() {
    return this.tracking_duration;
  }

  getJobType() {
    return this.job_type;
  }

  getRegistry() {
    return this.registry;
  }

  getProject() {
    return this.project;
  }

  getActivityType() {
    return this.activity_type;
  }

  getComments() {
    return this.comments;
  }

  getTask() {
    return this.task;
  }

  getCost() {
    return this.cost;
  }
}
